object.go

```diff
@@ -0,0 +1,284 @@
+// Copyright 2021 The NATS Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package nats
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/nats-io/nuid"
+)
+
+type ObjectStore interface {
+	// PutObject will place the contents from the reader into a new stream.
+	PutObject(cfg *ObjectConfig, reader io.Reader) error
+	// GetObject will pull the object from the stream and write to the writer.
+	GetObject(name string) (*ObjectResult, error)
+	// PutFile is convenience function to put a file into an object store.
+	PutFile(cfg *ObjectConfig, filename string) error
```

Comment:
Put/GetFile are simply conveniences on a name that is a path - I don't think they are needed.